From a9db168b2e70455282d03e85260688013c99d31e Mon Sep 17 00:00:00 2001
From: Fabian Thomas <fabian@fabianthomas.de>
Date: Fri, 16 Jun 2023 10:49:39 +0200
Subject: [PATCH 4/8] make DRAMsim3 configs path compatible with hammulator

---
 src/mem/DRAMsim3.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/mem/DRAMsim3.py b/src/mem/DRAMsim3.py
index bc7192c901..c1ad25dcf0 100644
--- a/src/mem/DRAMsim3.py
+++ b/src/mem/DRAMsim3.py
@@ -48,11 +48,11 @@ class DRAMsim3(AbstractMemory):
     )
 
     configFile = Param.String(
-        "ext/dramsim3/DRAMsim3/configs/" "DDR4_8Gb_x8_2400.ini",
+        "gem5/ext/dramsim3/DRAMsim3/configs/" "DDR4_8Gb_x8_2400.ini",
         "The configuration file to use with DRAMSim3",
     )
     filePath = Param.String(
-        "ext/dramsim3/DRAMsim3/", "Directory to prepend to file names"
+        "gem5/ext/dramsim3/DRAMsim3/", "Directory to prepend to file names"
     )
 
     def controller(self):
-- 
2.41.0

